### DISCUSSION TITLE
Data Model: Organize Treeview Relational Data

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#16444] [Archive]
@directus/team I'm currently working through the Tree View alias field and am finding it difficult to organize my nested items as desired.

There are two key questions:

**1. How do I custom sort tree view data?** 
A simple Drag n Drop looks possible, but won't take effect for me.

https://user-images.githubusercontent.com/62306630/201765042-285b56ad-830b-4330-8869-1eb2404f84ef.mp4

**2. Is it possible to return a "cleanly nested" object?**
If you make the call `https://example.directus.app/items/menu` then your items are flat, and you need to write a little algorithm to sort them.

<img width="1278" alt="CleanShot 2022-11-14 at 14 24 34@2x" src="https://user-images.githubusercontent.com/62306630/201765314-c2b1eaf7-30de-42ce-95a2-7cb9a068dd92.png">

If I try to call `https://example.directus.app/items/menu?fields=id,name,body,nest.*` I end up with repeat data... and its not guaranteed what my nested depth will be. So that approach seems unsuitable for this use case.

<img width="1278" alt="CleanShot 2022-11-14 at 15 10 54@2x" src="https://user-images.githubusercontent.com/62306630/201766289-85aebadc-c8e6-452c-89b1-0b7c972f55f1.png">

Not sure if its possible, but it'd be really cool if there was a way to nest Tree View data "on the fly" in the API call.

**Comment by rijkvanzanten** [comment]
1. Do you have a `sort` field set in the relationship? If so, drag and drop sort should work IIRC
2. This interface is "just" a o2m, so all the regular rules for a o2m apply here as well. You could achieve what you're looking for by adding `?filter[parent_id][_null]=true` and deeply nesting based on your depth rules with the dot notation in fields 👍🏻

**Reply by erondpowell** [reply]
1. Interestingly, looks like you can use a sort field... but looks like not on the item detail page. 

Should I open an issue for a UI improvement/bug-fix?

https://user-images.githubusercontent.com/62306630/201773798-745194be-7b46-4570-ba22-4e086481f104.mp4 

2. Trying this now.. Will report back if I can't get it to work.

**Reply by rijkvanzanten** [reply]
No need for a feature request for a tree layout, already exists 🙂  https://github.com/directus/directus/discussions/3054

**Reply by erondpowell** [reply]
> No need for a feature request for a tree layout, already exists 🙂 #3054

I meant about the ability to "drag n drop" nested siblings on the item detail page. Not a request for Tree Layout.

**Reply by erondpowell** [reply]
Notice how the UI *almost* lets me visually drag n drop, but won't apply the switch.

**Reply by rijkvanzanten** [reply]
ah gotcha
